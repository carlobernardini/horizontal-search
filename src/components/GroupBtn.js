import React from 'react';
import PropTypes from 'prop-types';
import { DISPLAY_NAME_GROUP_BTN } from '../constants';

const GroupBtn = ({ hidden, onClick }) => {
    if (hidden) {
        return null;
    }

    return (
        <button
            type="button"
            className="orizzonte__group-btn"
            onClick={ onClick }
        >
            &nbsp;
        </button>
    );
};

GroupBtn.displayName = DISPLAY_NAME_GROUP_BTN;

GroupBtn.propTypes = {
    hidden: PropTypes.bool,
    onClick: PropTypes.func
};

GroupBtn.defaultProps = {
    hidden: false,
    onClick: () => {}
};

export default GroupBtn;
