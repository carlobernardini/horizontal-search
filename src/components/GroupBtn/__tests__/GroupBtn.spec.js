import React from 'react';
import GroupBtn from '../GroupBtn';

describe('<GroupBtn />', () => {
    it('should render a default group button', () => {
        const onClick = jest.fn();
        const wrapper = shallow(
            <GroupBtn
                onClick={ onClick }
            />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });
    

    it('should render a hidden group button', () => {
        const wrapper = shallow(
            <GroupBtn
                hidden
            />
        );

        expect(wrapper).toMatchSnapshot();
    }); 
});
