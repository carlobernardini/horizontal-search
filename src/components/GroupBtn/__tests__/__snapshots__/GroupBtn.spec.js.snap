// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupBtn /> should render a default group button 1`] = `
<button
  className="orizzonte__group-btn"
  onClick={[MockFunction]}
  type="button"
>
  Â 
</button>
`;

exports[`<GroupBtn /> should render a hidden group button 1`] = `""`;
